### DISCUSSION TITLE
Please Remove Restrictions around ```required```

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#6745] [General]
This code should be legal but isn't:
```
    public class FooBase { 
        private required Dependency? Dep { get; init; }

        public FooBase Create() {
            var ret = new FooBase()
            {
                Dep = null,
            };

            return ret;
        }

    }
```

The visibility restrictions regarding ```required``` are artificially limiting my ability to write the code the way I want to.

**Comment by HaloFour** [comment]
See: https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-01-05.md#required-properties-accessibilities

I personally don't find this to be a terribly compelling scenario.  The `required` properties feature is intended to be used as guardrails for the consumer of the type, not the type itself.
